"New member joined your organization" email

Twixes opened the pull request:
## Changes

These Metabase emails got me inspired, I just could not resist. This adds a similar email that is sent to all organization members when a new member joins from an invite (obviously besides that new member themselves). Two benefits:
- if an unexpected (malicious?) user joins, people get notice
- people are nudged into coming back to PostHog, perhaps to discover something, driving retention

Sadly I have problems receiving the email locally despite everything seeming in order… mind taking a look @paolodamico?

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests

Twixes committed:
Update OrganizationInvite model to be marked used instead of deleted

Twixes committed:
Send email to other organization members when an invite is used

paolodamico reviewed (changes requested):
👏 100% on board with this! Will be very helpful in terms of security & engagement.

I'm thinking we could simplify the workflow a little bit here, let me know your thoughts. Instead of adding another attribute `used_at` and keeping all these invite records, we could simply fire a celery task (with proper retry handling) that contains the invite ID for use as `campaign_key`. I don't think there's a ton of value on keeping the invite records and adds more stuff to maintain.


In terms of the email here are my thoughts (ignore the missing images, `localhost` thing),
- I'm not sure about having the person who created the invite in there, think it may be leaking info out of PostHog, can even be interpreted as a bit accusatory, and the info is available in PostHog if you need it.
- I think the main CTA should be something more valuable on PostHog. For instance, we could prompt users to go to their dashboards to share some of them with the new member. At least I think this should just take you to PostHog home.
- Related to the above, we can change the members list to a secondary CTA. Something like, "To see the current member list for Hogflix Movies click here".
- Visually I think using the avatar thing in there (the one we use in the top navigation bar) would make it even more clear what's going on and who joined. 
- I'm thinking the subject line should be "Bob just joined you on PostHog" (i.e. remove the org name), think it might be too much for the subject line, and the line does read confusing at first.
- This is a more general comment for emails, but maybe worth adding something in the footer like: "This email is intended for {email} at Hogflix Movies".
<img width="680" alt="" src="https://user-images.githubusercontent.com/5864173/109428963-42fbca00-79ae-11eb-994a-f53441882243.png">

Twixes committed:
Merge branch 'master' into someone-else-joined-email

Twixes committed:
Update migration

Twixes committed:
Remove OrganizationInvite model changes

Twixes committed:
Address feedback

Twixes committed:
Update member_join.html

Twixes committed:
Fix authenticated org-less invite signup

Twixes committed:
Merge branch 'fix-authenticated-orgless-invite-signup' into someone-else-joined-email

Twixes committed:
Merge branch 'master' into someone-else-joined-email

Twixes committed:
Complete feature

Twixes commented:
Thanks a lot for the feedback! Finally got to prioritize this, so I addressed the points, added tests, and did some refactoring.

Replying to points:
1. Alright, removed inviting user from email.
2. Replaced the CTA with a general "View PostHog insights" one. Should be more valuable.
3. Didn't add a secondary CTA, I think we should keep this simple.
4. Didn't add the avatar. This _can_ be an improvement for a future PR, but it'd be additional work to get it in without React (which we rely on in the app), and ultimately it wouldn't be worth it as in most cases there'd be no Gravatar for a company email, leaving us with just the initial letter.
5. Simplified the subject line.
6. Appreciate the idea with adding the intended recipient email to the footer, but in this case particularly it's not feasible, as the email is mass sent. I don't think that'd we valuable enough to generate multiple personal versions really TBH.

Twixes commented:
This is what such an email looks like now:

![example](https://user-images.githubusercontent.com/4550621/118888274-bd781e00-b8fb-11eb-8a30-d983a433d286.png)

Twixes committed:
Merge branch 'master' into someone-else-joined-email

Twixes committed:
Add "just" to "joined you"

Twixes committed:
Merge branch 'master' into someone-else-joined-email

paolodamico commented on posthog/tasks/email.py:
```diff
@@ -125,3 +125,22 @@ def send_invite(invite_id: str) -> None:
+    )
+    # Don't send this email to the new member themselves
+    members_to_email = tuple(organization.members.exclude(email=invitee.email))
+    if members_to_email:
```
I don't think you need either the conversion to `tuple` or the check on whether it's truthy. If the queryset is empty, it will simply not add any recipients in the for loop.

paolodamico commented on posthog/templates/email/member_join.html:
```diff
@@ -0,0 +1,20 @@
+        {{ invitee.first_name }} (<a href="mailto:{{ invitee.email }}">{{ invitee.email }}</a>) just joined you at {{ organization.name }}.
+    </p>
+    <p>
+        PostHog is a product analytics platform made to help you understand your users.<br/>
```
Only comment: I would remove this line. If you're getting this message, you already know what PostHog is.

paolodamico reviewed (approved):
minor comments but lgtm!

Twixes committed:
Apply feedback

Twixes committed:
Merge branch 'master' into someone-else-joined-email
